shell/platform/linux/fl_key_event_plugin.cc

```diff
@@ -20,11 +24,16 @@ static constexpr char kUnicodeScalarValuesKey[] = "unicodeScalarValues";
 static constexpr char kGtkToolkit[] = "gtk";
 static constexpr char kLinuxKeymap[] = "linux";
 
+static constexpr uint64_t kMaxPendingEvents = 1000;
+
 struct _FlKeyEventPlugin {
   GObject parent_instance;
 
   FlBasicMessageChannel* channel = nullptr;
-  GAsyncReadyCallback response_callback = nullptr;
+  FlTextInputPlugin* text_input_plugin = nullptr;
+  FlKeyEventPluginCallback response_callback = nullptr;
+
+  std::deque<std::pair<uint64_t, GdkEventKey*>> pendingEvents;
```

Comment:
My initial impression is we should use a GLib container here, not a C++ style one (to avoid mixing styles too much). This is a bit harder though because we don't have a `std::pair` equivalent.

Reply:
Yeah, I thought about that too, but I didn't see any good data structure in the GObject docs. Do you have any suggestions?  I mean, I could make my own deque as a separate GObject class if you want. It's not terribly complex if all we want is a doubly-linked list.

Reply:
The go-to object I use is `GPtrArray` - it has the advantage over `GList` is in that you can set a function to free/unref the elements and avoid doing that manually.

The variable name should also be `pending_events`.

Reply:
OK, I converted to using a GPtrArray, and converted the pair into a new GObject class so that we're not mixing free and new.